### PR TITLE
Disable CodeCov expired reports validation

### PR DESCRIPTION
**Pull request by wzieba**
### Description

The majority of code coverage reports are considered incorrect by CodeCov (see [here](https://app.codecov.io/gh/woocommerce/woocommerce-android/commits?page=1)).

As valid reports exist (see e.g. #6241 ) I'm leaning toward the theory that the problem is with time-based validation from CodeCov ([docs](https://docs.codecov.com/docs/codecov-yaml#expired-reports)).

#### How CodeCov knows when a report has been generated?

At the beginning of the XML report, there's a piece of information about the sessions that created the report. Usually, there are two of them (because of two `.exec` files I suppose). There's timestamp for session creation and "dump" there. This is the only time-based information (I could find) that we send to CodeCov so I believe they take one of those values. You can find it by opening any failed report for e.g. commit, on the right under `Uploads` there's `Download` section to see payload. Then, look for `start="` phrase. 

If you convert those milliseconds to date, you can see that sometimes those point to a date a few days back. Also, I'm not sure if `session dump` means exactly the time of generating the report - I have a feeling that it's more like time of generating `.exec` file.

#### Why are the reports outdated?

Probably Gradle Build cache or cached up-to-date checks for generating `.exec` file (artifact from tests task). Just disabling cache for `jacocoTestReport` doesn't solve the issue. I'm not 100% sure how it works under the hood though, but I decided to timebox my time for that as it's still an experiment.

#### How does that influence CodeCov?

CodeCov by default doesn't accept reports than were generated 12h+ before sending payload. While this might work for other frameworks, I'm not sure if it's a good fit for JaCoCo (and Gradle).

I'm suggesting removing this validation - if our reports will happen to be invalid, we have a problem in our cache configuration and not with CodeCov.

See similar CodeCov issue: 
https://community.codecov.com/t/reports-are-not-displayed-there-was-an-error-processing-coverage-reports/750

### CODE REVIEW

**Comment by wzieba** [comment]
An update: I've asked the CodeCov team how they determine JaCoCo report creation time. Because if it's time of generating JaCoCo `.xml` file, maybe disabling cache/up-to-date checks for the task will resolve the issue.

https://community.codecov.com/t/how-codecov-knows-time-of-jacoco-report-generation/3571

For now, let's keep this PR as a draft 🙂.

**Comment by ParaskP7** [comment]
👋 @wzieba !

> An update: I've asked the CodeCov team how they determine JaCoCo report creation time. Because if it's time of generating JaCoCo .xml file, maybe disabling cache/up-to-date checks for the task will resolve the issue.

Thank you for this update! 👍 

> For now, let's keep this PR as a draft 🙂.

👍

PS: I'll be AFK from next week, for 2 weeks, so please don't wait on me to review this change, that is if you end up making it ready for review as some next week.

**Comment by wzieba** [comment]
Gotcha, thanks for the heads up @ParaskP7 ! 🙌

**Comment by wzieba** [comment]
We got the answer from the CodeCov team:

> Q: how CodeCov determines the time of JaCoCo report generation for the [“Expired Reports” 2](https://docs.codecov.com/docs/codecov-yaml#expired-reports) check?

> A: it is taken from all instances of sessioninfo > start. If any of those values are longer than the max_report_age, it will be considered expired.

I also checked how JaCoCo sessions are created and it looks like there's a separate session for each `.exec` file. Sessions have timestamps from `.exec` files and those files are cachable.

So what happens eventually is that when PR e.g. doesn't change anything in `cardreader` module, JaCoCo will use the cached `.exec` file of the module, which has an old timestamp. Then, CodeCov will drop this report because it's considered too old.

---

So, in conclusion, as we trust the Gradle Cache system for our tests, the more we should have it for code coverage. Hence, we don't need additional validation from CodeCov.

**Review by AliSoftware — APPROVED** [review]
Rationale in above discussion (especially last comments around trusting our cache) make sense to me 👍
